test/e2e/usersignup_test.go

```diff
@@ -45,6 +45,55 @@ func (s *userSignupIntegrationTest) TearDownTest() {
 	memberAwait2.Clean(s.T())
 }
 
+func (s *userSignupIntegrationTest) TestUserSignupMigration() {
+	hostAwait := s.Host()
+	memberAwait := s.Member1()
+	// Create user signup
+	hostAwait.UpdateToolchainConfig(s.T(),
+		testconfig.AutomaticApproval().Enabled(true),
+		testconfig.CapacityThresholds().
+			MaxNumberOfSpaces(
+				testconfig.PerMemberCluster(memberAwait.ClusterName, 1000),
+			).
+			ResourceCapacityThreshold(80))
+
+	// Create a new UserSignup
+	userSignup := NewUserSignup(hostAwait.Namespace, "migration-test@acme.com", "migration-test@acme.com")
+
+	// Clear the identity claims
+	userSignup.Spec.IdentityClaims = toolchainv1alpha1.IdentityClaimsEmbedded{}
+
+	// Set annotations
+	userSignup.Annotations[toolchainv1alpha1.SSOUserIDAnnotationKey] = "foo"
+	userSignup.Annotations[toolchainv1alpha1.SSOAccountIDAnnotationKey] = "bar"
+
+	err := hostAwait.CreateWithCleanup(s.T(), userSignup)
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' created", userSignup.Name)
+
+	identityClaimsSetCriteria := wait.UserSignupWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.UserSignup) bool {
+			return actual.Spec.IdentityClaims.Company == userSignup.Spec.Company &&
+				actual.Spec.IdentityClaims.Sub == userSignup.Name &&
+				actual.Spec.IdentityClaims.GivenName == userSignup.Spec.GivenName &&
+				actual.Spec.IdentityClaims.FamilyName == userSignup.Spec.FamilyName &&
+				actual.Spec.IdentityClaims.PreferredUsername == userSignup.Spec.Username &&
+				actual.Spec.IdentityClaims.UserID == "foo" &&
+				actual.Spec.IdentityClaims.AccountID == "bar" &&
+				actual.Spec.IdentityClaims.Email == userSignup.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey] &&
+				actual.Spec.IdentityClaims.OriginalSub == userSignup.Spec.OriginalSub
+
+		},
+		Diff: func(actual *toolchainv1alpha1.UserSignup) string {
+			return "expected UserSignup to have IdentityClaims migrated"
+		},
+	}
+
+	// Check the UserSignup is successfully migrated now
+	userSignup, err = hostAwait.WaitForUserSignup(s.T(), userSignup.Name, identityClaimsSetCriteria)
```

Comment:
maybe redundand - but should we also verify that the usersignup is approved and the provisioned resources?

something like:
```
	// Check the UserSignup is approved now
	userSignup, err = hostAwait.WaitForUserSignup(s.T(), userSignup.Name, wait.UntilUserSignupHasConditions(wait.ConditionSet(wait.Default(), wait.ApprovedAutomatically())...))
	require.NoError(s.T(), err)

	// Confirm the MUR was created and target cluster was set
	VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, userSignup, "deactivate30", "base")
```

Reply:
Good point but I think it's not necessary in this case.  If the migration condition test passes then the approval of the usersignup and provisioning are not a concern.